libs/langchain-community/src/chat_models/friendli.ts

```diff
@@ -0,0 +1,378 @@
+        message: {
+          role: FriendliMessageRole;
+          content: string;
+        };
+        finish_reason: string;
+      }[];
+      usage: {
+        prompt_tokens: number;
+        completion_tokens: number;
+        total_tokens: number;
+      };
+      created: number;
+    }
+
+    const messageList = messages.map((message) => {
+      if (typeof message.content !== "string") {
+        throw new Error(
+          "Friendli does not support non-string message content."
+        );
+      }
+      return {
+        role: messageToFriendliRole(message),
+        content: message.content,
+      };
+    });
+
+    const response = (await this.caller.call(async () =>
+      fetch("https://inference.friendli.ai/v1/chat/completions", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          Accept: "application/json",
+          Authorization: `Bearer ${this.friendliToken}`,
+          ...(this.friendliTeam
+            ? { "X-Friendli-Team": this.friendliTeam }
+            : {}),
+        },
+        body: JSON.stringify({
+          messages: messageList,
+          stream: false,
+          model: this.model,
+          max_tokens: this.maxTokens,
+          frequency_penalty: this.frequencyPenalty,
+          stop: this.stop,
+          temperature: this.temperature,
+          top_p: this.topP,
+        }),
+      }).then((res) => res.json())
+    )) as ChatFriendliResponse;
+
+    const generations: ChatGeneration[] = [];
+    for (const data of response.choices ?? []) {
+      const text = data.message?.content ?? "";
+      const generation: ChatGeneration = {
+        text,
+        message: friendliResponseToChatMessage(data.message ?? {}),
+      };
+      if (data.finish_reason) {
+        generation.generationInfo = { finish_reason: data.finish_reason };
+      }
+      generations.push(generation);
+    }
+
+    return { generations };
+  }
+
+  async *_streamResponseChunks(
+    messages: BaseMessage[],
+    _options: this["ParsedCallOptions"],
+    runManager?: CallbackManagerForLLMRun
+  ): AsyncGenerator<ChatGenerationChunk> {
+    interface ChatFriendliResponse {
+      choices: {
+        index: number;
+        delta: {
+          role: FriendliMessageRole;
+          content: string;
+        };
+        finish_reason: string;
+      }[];
+      created: number;
+    }
+
+    const messageList = messages.map((message) => {
+      if (typeof message.content !== "string") {
+        throw new Error(
+          "Friendli does not support non-string message content."
+        );
+      }
+      return {
+        role: messageToFriendliRole(message),
+        content: message.content,
+      };
+    });
+
+    const response = await this.caller.call(async () =>
+      fetch("https://inference.friendli.ai/v1/chat/completions", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          Accept: "application/json",
```

Comment:
This needs to be `Accept: "text/event-stream"`.

Reply:
Fixed to `Accept: "text/event-stream"`.